ui/types/vault/models/sync/destination.d.ts

```diff
@@ -0,0 +1,33 @@
+import type { WithFormFieldsAndValidationsModel } from 'vault/app-types';
+
+export default interface SyncDestinationModel extends WithFormFieldsAndValidationsModel {
+  name: string;
+  type: string;
+  get icon(): string;
+
+  // aws-sm
+  accessKeyId: string;
+  secretAccessKey: string;
+  region: string;
+
+  // azure-kv
+  keyVaultUri: string;
+  clientId: string;
+  clientSecret: string;
+  tenantId: string;
+  cloud: string;
+
+  // gcp
+  credentials: string;
+
+  // gh
+  accessToken: string;
+  repositoryOwner: string;
+  repositoryName: string;
+
+  // vercel project
+  accessToken: string;
+  projectId: string;
+  teamId: string;
+  deploymentEnvironments: array;
```

Comment:
These properties would need to be optional since they don't all exist based on the type.

Reply:
I wanted to ask you if I should separate these into separate models? It felt nicer in the form typescript file to just declare one `SyncDestinationModel` instead of list all the potential destinations and update it as they go